akka-persistence-testkit/src/main/scala/akka/persistence/testkit/scaladsl/UnpersistentBehavior.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.testkit.scaladsl
+
+import akka.actor.typed.Behavior
+import akka.persistence.testkit.internal.{ Unpersistent, PersistenceProbeImpl }
+
+import scala.reflect.ClassTag
+
+object UnpersistentBehavior {
+  /** Given an EventSourcedBehavior, produce a non-persistent Behavior which synchronously publishes events and snapshots
+   *  for inspection.  State is updated as in the EventSourcedBehavior, and side effects are performed synchronously.  The
+   *  resulting Behavior is, contingent on the command handling, event handling, and side effects being compatible with the
+   *  BehaviorTestKit, testable with the BehaviorTestKit.
+   *
+   *  The returned Behavior does not intrinsically depend on configuration: it therefore does not serialize and assumes an
+   *  unbounded stash for commands.
+   */
+  def fromEventSourced[Command, Event, State](
+      behavior: Behavior[Command],
+      fromStateAndOffset: Option[(State, Long)] = None): (Behavior[Command], PersistenceProbe[Event], PersistenceProbe[State]) = {
```

Comment:
The javadsl exposes a wrapper class instead of a `Tuple3`, though in the Java API, that exposes a "guaranteed-to-fail" probe for durable state.  Could get around that with 4 substantially-identical result types ("Java or Scala" x "Event-sourced or Durable-state"), but that strikes me as a "cure might be worse than the disease" situation